### PR TITLE
PADV-752  filters student page

### PR DESCRIPTION
**Pull request by AuraAlba**
### Ticket
https://agile-jira.pearson.com/browse/PADV-752

### Description
Change filters in student page according to new wireframe

### Changes made
Change filters component
Use components from paragon-topaz library
Add tests

### Screenshots
Screenshot from 2023-11-24 16-58-55

### How to test
Clone the Institution Portal MFE
Run npm install.
Run npm run start
Go to http://localhost:1980/students

### CODE REVIEW

File: src/features/Students/data/api.js [hunk 1]
```diff
@@ -23,7 +23,21 @@ function handleEnrollments(data, courseId) {
...
 
+function getCoursesByInstitution(institutionId) {
+  return getAuthenticatedHttpClient().get(
+    `${getConfig().COURSE_OPERATIONS_API_V2_BASE_URL}/courses/?institution_id=${institutionId}`,
+  );
+}
+
+function getClassesByInstitution(institutionId, courseName) {
+  return getAuthenticatedHttpClient().get(
+    `${getConfig().COURSE_OPERATIONS_API_V2_BASE_URL}/classes/?institution_id=${institutionId}?course_name=${courseName}`,
```
Comment (anfbermudezme) [inline src/features/Students/data/api.js:34]: Line too long.

File: src/index.scss [hunk 2]
```diff
@@ -3,4 +3,6 @@
 @import "@edx/brand/paragon/variables.scss";
 @import "@edx/paragon/scss/core/core.scss";
 @import "@edx/brand/paragon/overrides.scss";
-@import "react-paragon-topaz/src/DataTable/DataTable.scss";
\ No newline at end of file
+@import "react-paragon-topaz/src/DataTable/DataTable.scss";
+@import "react-paragon-topaz/src/Form/Form.scss";
+@import "react-paragon-topaz/src/Pagination/Pagination.scss";
```
Comment (anfbermudezme) [inline src/index.scss:8]: Add blank line.

File: src/features/Students/StudentsPage/index.scss [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    border: 1px solid #dfe1e1;
+    border-radius: 0.375rem;
+    padding: 1rem;
+}
+
+.page-content-container {
+    border: 1px solid #dfe1e1;
+    border-radius: 0.375rem;
+    padding: 20px 0;
+}
```
Comment (anfbermudezme) [inline src/features/Students/StudentsPage/index.scss:11]: Add blank line.

File: src/features/Students/StudentsFilters/index.jsx [hunk 4]
```diff
@@ -1,69 +1,219 @@
...
+  const [state, dispatch] = useReducer(reducer, initialState);
+  const [courseOptions, setCourseOptions] = useState([]);
+  const [classesOptions, setClassesOptions] = useState([]);
+  const [studentName, setStudentName] = useState('');
+  const [studentEmail, setStudentEmail] = useState('');
+  const [courseSelected, setCourseSelected] = useState(null);
+  const [classSelected, setClassSelected] = useState(null);
+  const [statusSelected, setStatusSelected] = useState(null);
+  const [examSelected, setExamSelected] = useState(null);
+  // check this after implementation of selector institution
```
Comment (anfbermudezme) [inline src/features/Students/StudentsFilters/index.jsx:46]: How will this change?

**Review by anfbermudezme — CHANGES_REQUESTED** [review]

File: src/features/Students/StudentsFilters/index.jsx (hunk 4 above)
Comment (AuraAlba) [reply src/features/Students/StudentsFilters/index.jsx:46]: The information of the institution selected won't be an array, just an object so we don't have to verify the length

**Review by anfbermudezme — APPROVED** [review]

**Pull request merged**